File: LeetCode/RemoveElement.js
let removeElement = function(array, val){

	let targetCount = array.filter(x => x == val).length;
	let non_targetCount = array.length - targetCount;
	let i = 0;
	let n = array.length;
	while (i < n){
		if(array[i] == val){
			array.splice(i, 1);
			n -= 1;
		}
		else{
			i += 1;
		}
	}
	// console.log(array);
	return non_targetCount;
};


const readline = require('readline')
const read = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

read.question('', function(arr){
	read.question('', function(val){
		arr = arr.split(" ");
		let temp = arr.map(function(value){
			return Number(value);
		});
		let result = removeElement(arr, val);
		console.log(result);
		read.close();
	});
});

read.on("close", function(){
	process.exit(0);
});

File: LeetCode/RemoveElement.cpp
#include<bits/stdc++.h>

using namespace std;

int removeElement(vector<int>& nums, int val) {
        int j = 0;
        for(int i=0; i<nums.size(); i++){
            if (nums[i] != val){
                nums[j++] = nums[i];
            }
        }
        return j;
}

int main(){
	int n;
	cin >> n;
	std::vector<int> v;
	int temp;
	for(int i=0; i<n; i++){
		cin >> temp;
		v.push_back(temp);
	}
	int value;
	cin >> value;
	int result = removeElement(v, value);
	cout << result<< endl;
	return 0;
}